### ISSUE TITLE
FVS: FVSIn screen changes

### ISSUE THREAD

**Issue by druepdx** [USFS-PNW/Fia-Biosum-Manager#372]
The following changes should be made to the FVSIn screen:

1. Remove the buttons and textboxes related to treatments and packages
2. Remove the package column from the grid. FVSIn.db files are created at the variant
level
3. Remove all columns related to FVSOut from the grid including the record counts
4. Move the FVSIn Options to the same page so that we no longer have two tabs

**Comment by jsfried** [comment]
That looks right. If it were possible to save the path from when the last data were loaded from an FIADB database in the DATABASE module to present here as the default "Path to FIA Datamart input SQLite database", that would be very helpful because many of us have multiple versions of very state FIADB database on our disks (from different years) and it is important to point to the same one from which the data were loaded in the database module. If the analyst changes that path here (perhaps because they moved the source database elsewhere on their computer, or because the path does not exist on their computer because the project  was formulated on another computer), it would be great if that choice stuck as the new default for the project.

**Comment by lbross** [comment]
We don't have a parameters database for FVSIn like we do for Processor and Optimizer. No settings from this screen are saved between runs. We could add this path to the application settings screen, like we have for the OpCost settings, but that would be out of scope for v5.11.2.
We could also be sure we are recording the path to the FIA database in the biosum_fvs_input_debug.txt log. I'm not sure how helpful that would be as it is overwritten every time FVSIn is created.

**Comment by jsfried** [comment]
Maybe the user should have a choice of whether to overwrite that log or append to it?

There is a configuration data file that is written to the variant folder under fvs\data when the input is made. It is also stored in the FVSIn.db

![image](https://github.com/user-attachments/assets/d5b52c9b-d312-4718-a282-14f2b1886430)
 
What we don’t have logs of is the DATABASE load. It would make sense to store this in master.db perhaps.

**Comment by lbross** [comment]
Since the Source FIA Data Mart database is saved in the FVSIn.db, it shouldn't be too hard to set 'Path to FIA Datamart input SQLite database' to this value. We'll need to check and see if the table exists in the FVSIn.db and if the path is still valid. @druepdx  please let us know if this is doable with the form redesign.
I will open a ticket to start saving the settings from the DATABASE load. This should wait until we move the database load to SQLite. We will need to map out which of the settings we want to save. This would only show the most recent load. Do you also want to write the configuration to a text file as FVSIn does? The nice thing about this is that it appends instead of overwriting so you have a full record.

**Comment by jsfried** [comment]
Agreed. Saving the config file to text (appended) is great docs to have...

**Comment by druepdx** [comment]
Here is a sample new screen layout
![image](https://github.com/user-attachments/assets/2be1b483-ab7a-4922-956a-bfcceb05fb84)

**Comment by jsfried** [comment]
Drue, this looks terrific. I have just a few suggestions:
1) do we need the asterisks in the tree growth calibration section? Couldn't they just read "include previous height, if available" and "include previous diameter, if available"?
2) "Include seedlings from FIADB" would avoid ambiguity (we also create seedlings with our kcp codes and this option does not affect that).
3) The Path labeling would be improved as: "Path to the SQLite database sourced from FIA Datamart". 
4) "Data Directory" as a label is ambiguous as to whether is is a data source or target (it's the latter). Why not: "Directory to write FVSIn.DB files to". It will require shrinking the path specification field a bit, but one can always scroll within that field if necessary.
5) When would the rerfresh button be needed? What will it do? Trees field might need to be a but wider depending on font size.
I'll need to rewrite the help screen (and users guide section) that go with this, so do let me know when it's finalized and I'll get started.

**Comment by druepdx** [comment]
I can definitely make all of those changes. I didn't want to make any changes other than layout changes at first, but all of those should be easy. As for the refresh button, it repopulates the variant selection table. I personally don't see when that would be use. I'm guessing if anything would change with that button, the analyst would have done something outside of the FVSIn screen and everything on it would be reloaded when they reopen it.

**Comment by lbross** [comment]
My guess is that the refresh button is an artifact of when this screen was used for FVSOut. It probably updated all of the record counts that we've eliminated. It can go.

**Comment by jsfried** [comment]
Looks terrific!

**Comment by druepdx** [comment]
As long as there are records in the plot table, variants should be showing in that box. As long as we are using the same version of the CWI_demo project, EC and WC should both be showing.

**Comment by Cat-Ducat-USDA** [comment]
> As long as there are records in the plot table, variants should be showing in that box. As long as we are using the same version of the CWI_demo project, EC and WC should both be showing.

Gotcha. I'm going to test it again on this new project I built. Also wanted to add that the variant folders are still being created, but are empty. The FVSIn is combined, though.

**Comment by lbross** [comment]
@druepdx  let's stop creating the variant folders, unless we need them for a reason I'm not aware of. @Cat-Ducat-USDA keep us posted on the variant listing.

**Comment by druepdx** [comment]
Good catch! I'll track down where those are created and get rid of them.

**Comment by jsfried** [comment]
The variant selection boxes are popping up for me just fine. The simplified one-tab screen looks great. When I went to create the FVSIn though (overwriting the one in my existing project), BioSum returned: 

![Image](https://github.com/user-attachments/assets/f2ded1ac-9ff7-4823-9c7b-a05be944c9ae)

Cat did you have that experience? I did turn on the inclusion of surface fuel model field crew call (which is not the default)-- don't know if that is related.

**Comment by Cat-Ducat-USDA** [comment]
So the variant check boxes are popping up for me now in my new project I am building (for validation purposes - the one cycle project) and on my cwi_demo backup... so I am not sure what happened or why they were missing the first time around, but I have them now. Jeremy, yes, I did experience that the first time I tried to create the FVS input file, but after closing out of and restarting BioSum it proceeded fine.

**Comment by Cat-Ducat-USDA** [comment]
@druepdx I was using the most recent version (backup) of the cwi_demo project I provided to y'all I'm fairly sure, will check on that, but it is working now.

**Comment by druepdx** [comment]
Can you check the FVSIn.db and see if the treeinit and standinit tables have a column named fvs_variant?

**Comment by Cat-Ducat-USDA** [comment]
@druepdx yes, in my treeinit cond table, but not standinit table. So I need to remove that? Guess I'm using the backup where I did not remove that column yet.

**Comment by druepdx** [comment]
Yes, that's what's causing the issue! You can just manually delete that column if you don't want to start over your testing with a new project. That column won't exist in any other projects, so it shouldn't cause an issue. I also double checked the most recent version of CWI_demo (from 12/19) and it does not have that column, so the most recent sample project is up to date.

**Comment by Cat-Ducat-USDA** [comment]
Phew, thanks so much, Drue! Deleting that column presently - I was 90% sure I'd removed it from the copy I gave y'all so I'm glad that is the case!

**Comment by jsfried** [comment]
From what table does it need to be removed?

**Comment by druepdx** [comment]
It needs to be removed from the FVS_TREEINIT_COND table

**Comment by jsfried** [comment]
In what database? IN what folder? I keep deleting the FVSIN.DB in the FVS\data folder and it keeps making a new one with the offending column when I attempt to create a new FVSIn

**Comment by jsfried** [comment]
That took a lot of reruns to get right! So the issues were that the FVSIn.DB in the variant specific folder (EC) of the existing CWI_demo project had an extra column (fvs_variant) as the last column in the table (was not the case for the WC variant).  Even though BioSum was creating a new FVSIn.DB in the fvs\data folder (for all variants), it was affected by the variables present in the version of the fvsin.db in the specific variant folder. Moreover, when I removed the extra column I STILL got the error because the FVSIn.DB already created (in a failed attempt in the fvs\data folder) has a treeinit table with the offending variable that then affected the new FVSIn.DB being created. Had to DELETE that FVSIN.DB to remove all traces of the extra column to get to a run without errors.

**Comment by Cat-Ducat-USDA** [comment]
Hello all, I created another new project with the new msi, and the variant folders are GONE! Success! Thanks, Drue :)
